lib/rust/ensogl/component/text/src/font.rs

```
@@ -0,0 +1,617 @@
+}
+
+/// A variable font family. Contains font family definition and the font face.
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub struct VariableFamily {
+    pub definition: VariableFamilyDefinition,
+    pub face:       Rc<RefCell<Option<Face>>>,
+    pub last_axes:  Rc<RefCell<Option<VariationAxes>>>,
+}
+
+impl NonVariableFamily {
+    /// Load all font faces from the embedded font data. Corrupted faces will be reported and
+    /// ignored.
+    fn load_all_faces(&self, embedded: &Embedded) {
+        for (header, file_name) in &self.definition.map {
+            if let Some(face) = Face::load_from_memory(&*file_name, embedded) {
+                self.faces.borrow_mut().insert(*header, face);
+            }
+        }
+    }
+}
+
+impl VariableFamily {
+    /// Load all font faces from the embedded font data. Corrupted faces will be reported and
+    /// ignored.
+    fn load_all_faces(&self, embedded: &Embedded) {
+        if let Some(face) = Face::load_from_memory(&self.definition.file_name, embedded) {
+            self.face.borrow_mut().replace(face);
+        }
+    }
+}
+
+impl Family<NonVariableFaceHeader> for NonVariableFamily {
+    fn update_msdfgen_variations(&self, _variations: &NonVariableFaceHeader) {}
+    fn with_borrowed_face<F, T>(&self, variations: &NonVariableFaceHeader, f: F) -> Option<T>
+    where F: for<'a> FnOnce(&'a Face) -> T {
+        self.faces.borrow().get(variations).map(f)
+    }
+}
+
+impl Family<VariationAxes> for VariableFamily {
+    fn update_msdfgen_variations(&self, variations: &VariationAxes) {
+        if let Some(face) = self.face.borrow().as_ref() {
+            if self.last_axes.borrow().as_ref() != Some(&variations) {
+                self.last_axes.borrow_mut().replace(variations.clone());
+                variations.with_standard_axes(|axis| {
+                    let value = axis.value.into_inner() as f64;
+                    face.msdf.set_variation_axis2(axis.tag, value).ok();
```

Someone could possibly create Variable face (from ttf for example) from Non-variable, and in that case silencing an error is a bad idea.

Even if it would be impossible, we could log an error just in case.